typeutils/unsafe.go

```diff
@@ -11,20 +11,24 @@ import (
 // passed as argument are changed while the string is used.  No checking whether
 // bytes are valid UTF-8 data is performed.
 func BytesToString(b []byte) string {
+	// ensure the underlying bytes don't get GC'ed before the assignment happens
+	defer runtime.KeepAlive(&b)
+
 	return *(*string)(unsafe.Pointer(&b))
 }
 
 // Converts a string into a slice of bytes without performing a copy.
 // NOTE: This is an unsafe operation and may lead to problems if the bytes are changed.
 func StringToBytes(s string) []byte {
+	// ensure the underlying string doesn't get GC'ed before the assignment happens
+	defer runtime.KeepAlive(&s)
```

Comment:
I am all for `defer` :slightly_smiling_face: 
However, in the documentation it says:
`This ensures that the object is not freed, and its finalizer is not run, before the point in the program where KeepAlive is called. `

Even if the call of `defer` qualifies as a "physical point" at the very end of the function. This is still very hard to understand, especially with the way deferred function parameters are treated in go.
Is there a particular motivation for `defer`? The `unsafe.Pointer` will never panic so it should not even make a difference, if I understand correctly.